tests/integration/test_analysisd/test_rootcheck/test_configuration/test_rootcheck_delete.py

```diff
@@ -0,0 +1,155 @@
+                   daemons_handler, wait_for_rootcheck_start, truncate_monitored_files,
+                   simulate_agents):
+    '''
+    Testing with daemons_handler,
+    description: Check if the 'rootcheck' modules is working properly, that is, by checking if the logs
+                 are deleted correctly.
+                 For this purpose, the test will create a specific number of agents, and enable the rootcheck module.
+                 The rootcheck events will be sent for 60 seconds. After the time has passed, the rootcheck module
+                 gets disabled and check if the logs are deleted from the database when sending the delete
+                 table request.
+
+    wazuh_min_version: 4.2.0
+
+    tier: 0
+
+    parameters:
+        - test_configuration:
+            type: dict
+            brief: Configuration loaded from `configuration_templates`.
+        - test_metadata:
+            type: dict
+            brief: Test case metadata.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Apply changes to the ossec.conf configuration.
+        - wait_for_rootcheck_startup:
+            type: fixture
+            brief: Wait until the 'wazuh-analysisd' has begun and the 'alerts.json' file is created.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate wazuh logs.
+        - daemons_handler:
+            type: fixture
+            brief: Handler of Wazuh daemons.
+        - simulate_agents:
+            type: fixture
+            brief: Handler simulates agents.
+
+    assertions:
+        - Verify that the rootcheck events are deletet from the database
+    input_description: Different test cases are contained in an external YAML file (cases_configuration.yaml)
+                       which includes configuration settings for the 'rootcheck' module.
+    expected_output:
+        - Wazuh DB returned an error trying to delete the agent
+        - Rootcheck events were not deleted
+    '''
+    injectors = []
+    agents = simulate_agents
+
+    for agent in agents:
+        agent.modules['rootcheck']['status'] = 'enabled'
+        _, injector = connect(agent)
+        injectors.append(injector)
+        agents.append(agent)
+
+    # Let rootcheck events to be sent for 60 seconds
+    time.sleep(60)
+
+    for injector in injectors:
+        injector.stop_receive()
+
+    # Service needs to be restarted
+    control_service('start')
+
+    for agent in agents:
+        response = send_request_socket(f'agent {agent.id} rootcheck delete')
+        assert response.startswith(b'ok'), "Wazuh DB returned an error " \
+                                            "trying to delete the agent"
+
+    # Wait 5 seconds
+    time.sleep(5)
+
+    # Service needs to be stopped otherwise db lock will be held by Wazuh db
+    control_service('stop')
+
+    # Check that logs have been deleted
+    for agent in agents:
+        rows = get_sqlite_query_result(
```

Comment:
Instead of this, I think it's better to create a function in the framework in `utils/db_queries/agent_db.py`. Check this example: 
Framework: https://github.com/wazuh/qa-integration-framework/pull/14/files#diff-8e6468ba5213e65da741a71e177bc574f283eb6ea2235141bcafda50ae1e88d8R270
Tests: https://github.com/wazuh/wazuh/pull/18567/files#diff-025a8ddc35658fdb909e2da9fd4e799a3316e32879273ae3354419ebc23bf2f7R59

Reply:
If you use the function `query_wdb` (in the framework wrapper), you won't need `get_sqlite_query_result`. Query should be something like this:
```
agent {agent_id} sql SELECT * FROM pm_event
```
Use the wrapper here, the query should be in the framework.